generator/spring-boot/src/main/java/io/fabric8/maven/generator/springboot/SpringBootGenerator.java

```diff
@@ -52,6 +60,40 @@ public boolean isApplicable(List<ImageConfiguration> configs) {
     }
 
     @Override
+    public List<ImageConfiguration> customize(List<ImageConfiguration> configs) throws MojoExecutionException {
+        // if enabled lets generate a spring devtools token
+        String flag = getProjectProperty("fabric8.spring-devtools.generateToken", null);
+        boolean generateToken = !Strings.isNullOrEmpty(flag) && Configs.asBoolean(flag);
+        if (generateToken || getContext().runningWithGoal("fabric8:watch-spring-boot", "fabric8:watch")) {
+            generateSpringDevToolsToken();
+        }
+        return super.customize(configs);
+    }
+
+    private void generateSpringDevToolsToken() throws MojoExecutionException {
+        Properties properties = MavenUtil.getSpringBootApplicationProperties(getProject());
+        String remoteSecret = properties.getProperty(DEV_TOOLS_REMOTE_SECRET);
+        if (Strings.isNullOrEmpty(remoteSecret)) {
+            String newToken = UUID.randomUUID().toString();
+            log.info("Generating the spring devtools token: " + newToken);
+
+            File file = new File(getProject().getBasedir(), "target/classes/application.properties");
+            file.getParentFile().mkdirs();
+
+            String text = "# lets configure the spring devtools remote secret\nspring.devtools.remote.secret=" + newToken + "\n";
+
+            if (file.exists()) {
+                text = "\n" + text;
+            }
+            try (FileWriter writer = new FileWriter(file, true)) {
+                writer.append(text);
+            } catch (IOException e) {
```

Comment:
I would simply throw the IOException here (adapting the customize() interface). That way we dont need to add later a catch on `Exception` and simply can leave the catch on `IOException`.